epidose/device/util.sh

```diff
@@ -129,6 +136,53 @@ run_python()
   "$PYTHON" "$SCRIPT_DIR"/"$prog".py $DEBUG_FLAG $VERBOSE_FLAG "$@"
 }
 
+# Check the validity of the cuckoo filter
+# preconditions: none 
+# Outputs, through the stdout,
+# the time interval until a Cuckoo filter is valid (in seconds)
+# If 0 is returned, the Cuckoo filter is stale.
+get_filter_validity_age()
+{
+  if [ -r "$FILTER" ] ; then
+    filter_mtime=$(stat -c '%Y' "$FILTER")
+    time_now=$(date +%s)
+    filter_age=$((time_now - filter_mtime))
+    if [ $filter_age -lt $MAX_FILTER_AGE ] ; then
+      log "Filter's age ($filter_age s) is current; no update required"
+      validity_time=$((MAX_FILTER_AGE - filter_age))
+      echo "$validity_time"
+    else
+      log "Filter's age ($filter_age s) makes it stale; update required"
+      validity_time=0
+      echo "$validity_time"
+    fi
+  else
+    log "No filter available; download required"
+    mkdir -p "$(dirname $FILTER)"
+  fi	
+}
+
+# Obtain a (new) version of the Cuckoo filter
+# preconditions: none 
+# Returns an exit code that defines
+# whether a request to the ha-sever, to fetch a new cuckoo filter,
+# was successful.
+# If 1 is returned, then cuckoo filter was not obtained.
+get_new_filter()
+{
+  if curl --silent --show-error --fail --output "$FILTER.new" \
```

Comment:
Please use the previours `err=$(..)` construct to obtain and log curl's output.

Reply:
Ok, done.
I have updated my commit.